pkg/actors/placement/placement_test.go

```diff
@@ -425,17 +425,15 @@ func (s *testServer) ReportDaprStatus(srv placementv1pb.Placement_ReportDaprStat
 			return status.Error(codes.FailedPrecondition, "only leader can serve the request")
 		}
 
-		req, err := srv.Recv()
-		if err == io.EOF {
```

Comment:
This was the core of the issue. Sometimes in the test server (only in test code) the shutdown was reported as a gRPC "Context canceled" error, rather than just EOF. This seems to be a bug in the test code only, as the app code handles this case already.